engines/director/lingo/lingo-code.cpp

```diff
@@ -226,6 +226,9 @@ void LC::c_printtop(void) {
 	case OBJECT:
 		warning("#%s", d.u.s->c_str());
 		break;
+	case ARRAY:
+		warning("ARRAY: %s", d.toString()->c_str());
```

Comment:
That should be just toString(). The thing is that I am not sure that I implemented toString() for arrays correctly. Somebody needs to test that. And here it should produce the output as it is now, e.g. "[<el1>, <el2>. ..]"

Reply:
This array: ["one", "two", "three"] is printed as: WARNING: ARRAY: [one, two, three]! 
Without the `->c_str()` part its: WARNING: ARRAY: !

The `toString()` works correctly when showing the debug stack after message: 
Stack after: <[one, two, three]>

What would you like to see? A re-implementation of toString() for Arrays, or a custome print function in this code?

Reply:
Sorry for the created cofusion. I meant:
```
warning("%s", d.toString()->c_str());
```